### ISSUE TITLE
Setting ranges on Dimensions fails for Spread

### ISSUE THREAD

**Issue by mrksr** [ioam/holoviews#1445]
Usually, it is possible to explicitly supply ranges for plots using `redim`:
```python
hv.Overlay([
    hv.Curve(range(3)),
    hv.Curve(range(1, 4)),
]).redim(y={'range': (-2, 2)})
```
This results in a plot with the y axis ranging from -2 to 2 in matplotlib or sets the default view in bokeh, regardless of the actual value range of the data.

Doing the same for plots including a `Spread` fails:
```python
hv.Overlay([
    hv.Curve(range(3)),
    hv.Spread((range(3), range(3), [0.5, 1, 5]))
]).redim(y={'range': (-2, 2)}, yerror={'range': (-2, 2)})
```
The ranges behave as "minimum extents" for the axis but can grow beyond that.
Here is a screenshot which shows the correct behaviour (above) and the incorrect behaviour (below). Both y-axes should be ranging from -2 to 2.
![spread_range](https://cloud.githubusercontent.com/assets/5184063/26103807/767fc6da-3a3b-11e7-8b0b-f6254845cf13.png)

According to @philippjfr, the problem could also exist for `Histogram` and `ErrorBars`.

**Comment by philippjfr** [comment]
This should be fixed by #1446, and will be available in the 1.8.0dev1 release happening shortly.

**Comment by philippjfr** [comment]
Btw you can simplify your examples to:

```python
hv.Overlay([
    hv.Curve(range(3)),
    hv.Spread((range(3), range(3), [0.5, 1, 5]))
]).redim.range(y=(-2, 2), yerror=(-2, 2))
```
